File: test_mcp_connector_no_auth.py
#!/usr/bin/env python3
"""
Test script for Anthropic MCP Connector
Tests the MCP connector feature against a remote time server using direct HTTP requests.
"""

import os
import json
import requests
from dotenv import load_dotenv

def main():
    # Load environment variables
    load_dotenv()
    
    # Get API key
    api_key = os.getenv('ANTHROPIC_API_KEY')
    if not api_key:
        raise ValueError("ANTHROPIC_API_KEY not found in environment variables")
    
    # Read MCP server URL from mcp_servers.json
    with open('mcp_servers.json', 'r') as f:
        server_url = f.read().strip()
    
    print(f"Testing MCP Connector with server: {server_url}")
    print("=" * 60)
    
    # Configure MCP server
    mcp_servers = [
        {
            "type": "url",
            "url": server_url,
            "name": "time-server"
        }
    ]
    
    # API endpoint and headers
    url = "https://api.anthropic.com/v1/messages"
    headers = {
        "Content-Type": "application/json",
        "X-API-Key": api_key,
        "anthropic-version": "2023-06-01",
        "anthropic-beta": "mcp-client-2025-04-04"
    }
    
    # Test 1: Ask what tools are available
    print("\n🔍 Test 1: Discovering available tools")
    print("-" * 40)
    
    payload1 = {
        "model": "claude-sonnet-4-20250514",
        "max_tokens": 1000,
        "messages": [
            {"role": "user", "content": "What tools do you have available from the MCP servers?"}
        ],
        "mcp_servers": mcp_servers
    }
    
    response1 = requests.post(url, headers=headers, json=payload1)
    
    if response1.status_code == 200:
        result1 = response1.json()
        print("Response:", result1["content"][0]["text"])
    else:
        print(f"Error {response1.status_code}: {response1.text}")
        return
    
    # Test 2: Get current time
    print("\n⏰ Test 2: Getting current time")
    print("-" * 40)
    
    payload2 = {
        "model": "claude-sonnet-4-20250514",
        "max_tokens": 1000,
        "messages": [
            {"role": "user", "content": "What is the current time in Austin, TX? Use the time server tools."}
        ],
        "mcp_servers": mcp_servers
    }
    
    response2 = requests.post(url, headers=headers, json=payload2)
    
    if response2.status_code == 200:
        result2 = response2.json()
        # Display response content including tool use blocks
        for content_block in result2["content"]:
            if content_block["type"] == "text":
                print("Text response:", content_block["text"])
            elif content_block["type"] == "mcp_tool_use":
                print(f"🔧 MCP Tool Use:")
                print(f"  - Tool: {content_block['name']}")
                print(f"  - Server: {content_block['server_name']}")
                print(f"  - Input: {content_block['input']}")
            elif content_block["type"] == "mcp_tool_result":
                print(f"📊 MCP Tool Result:")
                print(f"  - Tool Use ID: {content_block['tool_use_id']}")
                print(f"  - Is Error: {content_block['is_error']}")
                print(f"  - Content: {content_block['content']}")
    else:
        print(f"Error {response2.status_code}: {response2.text}")
        return
    
    # Test 3: Convert time between timezones
    print("\n🌍 Test 3: Converting time between timezones")
    print("-" * 40)
    
    payload3 = {
        "model": "claude-sonnet-4-20250514",
        "max_tokens": 1000,
        "messages": [
            {"role": "user", "content": "Convert the current time from America/New_York to America/Chicago (Austin, TX timezone) using the time server tools."}
        ],
        "mcp_servers": mcp_servers
    }
    
    response3 = requests.post(url, headers=headers, json=payload3)
    
    if response3.status_code == 200:
        result3 = response3.json()
        # Display response content
        for content_block in result3["content"]:
            if content_block["type"] == "text":
                print("Text response:", content_block["text"])
            elif content_block["type"] == "mcp_tool_use":
                print(f"🔧 MCP Tool Use:")
                print(f"  - Tool: {content_block['name']}")
                print(f"  - Server: {content_block['server_name']}")
                print(f"  - Input: {content_block['input']}")
            elif content_block["type"] == "mcp_tool_result":
                print(f"📊 MCP Tool Result:")
                print(f"  - Tool Use ID: {content_block['tool_use_id']}")
                print(f"  - Is Error: {content_block['is_error']}")
                print(f"  - Content: {content_block['content']}")
    else:
        print(f"Error {response3.status_code}: {response3.text}")
        return
    
    print("\n✅ MCP Connector testing completed!")

if __name__ == "__main__":
    main()

File: test-mcp-connector_no_auth.ts
#!/usr/bin/env tsx
/**
 * Test script for Anthropic MCP Connector (TypeScript)
 * Tests the MCP connector feature against a remote time server.
 */

import fs from 'fs';
import { config } from 'dotenv';
import Anthropic from '@anthropic-ai/sdk';

// Load environment variables
config();

interface MCPServer {
  type: 'url';
  url: string;
  name: string;
  authorization_token?: string;
}

interface MCPToolUse {
  type: 'mcp_tool_use';
  id: string;
  name: string;
  server_name: string;
  input: Record<string, any>;
}

interface MCPToolResult {
  type: 'mcp_tool_result';
  tool_use_id: string;
  is_error: boolean;
  content: Array<{ type: string; text?: string; [key: string]: any }>;
}

interface TextContent {
  type: 'text';
  text: string;
}

type ContentBlock = TextContent | MCPToolUse | MCPToolResult;

async function main(): Promise<void> {
  // Get API key
  const apiKey = process.env.ANTHROPIC_API_KEY;
  if (!apiKey) {
    throw new Error('ANTHROPIC_API_KEY not found in environment variables');
  }

  // Initialize Anthropic client
  const anthropic = new Anthropic({
    apiKey: apiKey,
    defaultHeaders: {
      'anthropic-beta': 'mcp-client-2025-04-04',
    },
  });

  // Read MCP server URL from mcp_servers.json
  const serverUrl = fs.readFileSync('mcp_servers.json', 'utf8').trim();

  console.log(`Testing MCP Connector with server: ${serverUrl}`);
  console.log('='.repeat(60));

  // Configure MCP server
  const mcpServers: MCPServer[] = [
    {
      type: 'url',
      url: serverUrl,
      name: 'time-server',
    },
  ];

  // Test 1: Ask what tools are available
  console.log('\n🔍 Test 1: Discovering available tools');
  console.log('-'.repeat(40));

  try {
    const response1 = await anthropic.messages.create({
      model: 'claude-sonnet-4-20250514',
      max_tokens: 1000,
      messages: [
        {
          role: 'user',
          content: 'What tools do you have available from the MCP servers?',
        },
      ],
      mcp_servers: mcpServers,
    } as any); // Type assertion needed for beta feature

    const textContent = response1.content.find(
      (block): block is TextContent => block.type === 'text'
    );
    if (textContent) {
      console.log('Response:', textContent.text);
    }
  } catch (error) {
    console.error('Error in Test 1:', error);
    return;
  }

  // Test 2: Get current time
  console.log('\n⏰ Test 2: Getting current time');
  console.log('-'.repeat(40));

  try {
    const response2 = await anthropic.messages.create({
      model: 'claude-sonnet-4-20250514',
      max_tokens: 1000,
      messages: [
        {
          role: 'user',
          content: 'What is the current time? Use the time server tools.',
        },
      ],
      mcp_servers: mcpServers,
    } as any);

    // Display response content including tool use blocks
    for (const contentBlock of response2.content as ContentBlock[]) {
      if (contentBlock.type === 'text') {
        console.log('Text response:', contentBlock.text);
      } else if (contentBlock.type === 'mcp_tool_use') {
        console.log('🔧 MCP Tool Use:');
        console.log(`  - Tool: ${contentBlock.name}`);
        console.log(`  - Server: ${contentBlock.server_name}`);
        console.log(`  - Input:`, contentBlock.input);
      } else if (contentBlock.type === 'mcp_tool_result') {
        console.log('📊 MCP Tool Result:');
        console.log(`  - Tool Use ID: ${contentBlock.tool_use_id}`);
        console.log(`  - Is Error: ${contentBlock.is_error}`);
        console.log(`  - Content:`, contentBlock.content);
      }
    }
  } catch (error) {
    console.error('Error in Test 2:', error);
    return;
  }

  // Test 3: Convert time between timezones
  console.log('\n🌍 Test 3: Converting time between timezones');
  console.log('-'.repeat(40));

  try {
    const response3 = await anthropic.messages.create({
      model: 'claude-sonnet-4-20250514',
      max_tokens: 1000,
      messages: [
        {
          role: 'user',
          content:
            'Convert the current time from America/New_York to America/Chicago (Austin, TX timezone) using the time server tools.',
        },
      ],
      mcp_servers: mcpServers,
    } as any);

    // Display response content
    for (const contentBlock of response3.content as ContentBlock[]) {
      if (contentBlock.type === 'text') {
        console.log('Text response:', contentBlock.text);
      } else if (contentBlock.type === 'mcp_tool_use') {
        console.log('🔧 MCP Tool Use:');
        console.log(`  - Tool: ${contentBlock.name}`);
        console.log(`  - Server: ${contentBlock.server_name}`);
        console.log(`  - Input:`, contentBlock.input);
      } else if (contentBlock.type === 'mcp_tool_result') {
        console.log('📊 MCP Tool Result:');
        console.log(`  - Tool Use ID: ${contentBlock.tool_use_id}`);
        console.log(`  - Is Error: ${contentBlock.is_error}`);
        console.log(`  - Content:`, contentBlock.content);
      }
    }
  } catch (error) {
    console.error('Error in Test 3:', error);
    return;
  }

  console.log('\n✅ MCP Connector testing completed!');
}

// Run the main function
main().catch((error) => {
  console.error('Fatal error:', error);
  process.exit(1);
});
